HX711.cpp

```diff
@@ -44,50 +44,48 @@ void HX711::set_gain(byte gain) {
 }
 
 long HX711::read() {
-	// wait for the chip to become ready
-	while (!is_ready()) {
-		// Will do nothing on Arduino but prevent resets of ESP8266 (Watchdog Issue)
-		yield();
-	}
-
-    unsigned long value = 0;
-    byte data[3] = { 0 };
-    byte filler = 0x00;
+    // Byte:     0        1        2        3
+    // Bits:  76543210 76543210 76543210 76543210
+    // Data: |--------|--------|--------|--------|
+    // Bit#:  33222222 22221111 11111100 00000000
+    //        10987654 32109876 54321098 76543210
+    union DataBuffer {
+        byte data[4];
+        long value;
+    } data_buffer;
+
+    // Wait for the chip to become ready
+    for (; !is_ready() ;) {
+        // Will do nothing on Arduino but prevent resets of ESP8266 (Watchdog Issue)
+        yield();
+    }
 
-	// pulse the clock pin 24 times to read the data
-    data[2] = shiftIn(DOUT, PD_SCK, MSBFIRST);
-    data[1] = shiftIn(DOUT, PD_SCK, MSBFIRST);
-    data[0] = shiftIn(DOUT, PD_SCK, MSBFIRST);
+    // Pulse the clock pin 24 times to read the data
+    data_buffer.data[1] = shiftIn(DOUT, PD_SCK, MSBFIRST);
+    data_buffer.data[2] = shiftIn(DOUT, PD_SCK, MSBFIRST);
+    data_buffer.data[3] = shiftIn(DOUT, PD_SCK, MSBFIRST);
 
-	// set the channel and the gain factor for the next reading using the clock pin
-	for (unsigned int i = 0; i < GAIN; i++) {
+	// Set the channel and the gain factor for the next reading using the clock pin
+	for (unsigned int i = GAIN ; 0 < i ; --i) {
```

Comment:
This is premature optimisation. Compiler is clever enough to see you don't use iteration variables inside loop body and will find best loop variant for you.